### ISSUE TITLE
Found with fuzzer: "timer sanity: melt timer 1386 on non-ice 25 <3,5>"

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#23]
```

Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7365859 in __GI_abort () at abort.c:79
#2  0x0000555555a44e78 in NH_abort () at end.c:236
#3  0x0000555555a4bc74 in panic (str=0x555556523120 "%s") at end.c:802
#4  0x0000555555f99ce3 in impossible (
    s=0x555556580900 "timer sanity: melt timer %ld on non-ice %d <%d,%d>") at pline.c:518
#5  0x00005555561b9c09 in timer_sanity_check () at timeout.c:2402
#6  0x000055555585f746 in sanity_check () at cmd.c:4843
#7  0x0000555555763767 in moveloop (resuming=0 '\000') at allmain.c:767
#8  0x00005555563bb426 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353


#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
        set = {__val = {0, 0, 0, 0, 0, 140737337632192, 93824997411658, 335544320, 
            140737341055120, 1073741824, 0, 140737342801568, 4294967295, 140737337360384, 
            140737342784672, 5886922055288131072}}
        pid = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
#1  0x00007ffff7365859 in __GI_abort () at abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0xffffffff, sa_sigaction = 0xffffffff}, 
          sa_mask = {__val = {140737337360384, 140737342784672, 5886922055288131072, 
              140737488345312, 140737342801568, 18446744073709547520, 140737327112192, 
              140737345470680, 140737488346384, 140737488346384, 140737344339986, 
              17592186043164, 140737488338944, 140737341055120, 5886922055288131072, 
              140737488345712}}, sa_flags = 1172082176, sa_restorer = 0x7fffffffdc30}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x0000555555a44e78 in NH_abort () at end.c:236
        gdb_prio = 1
        libc_prio = 2
        aborting = 1 '\001'
#3  0x0000555555a4bc74 in panic (str=0x555556523120 "%s") at end.c:802
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7fffffffdd00, 
            reg_save_area = 0x7fffffffdc30}}
#4  0x0000555555f99ce3 in impossible (
    s=0x555556580900 "timer sanity: melt timer %ld on non-ice %d <%d,%d>") at pline.c:518
        pbuf = "timer sanity: melt timer 1386 on non-ice 25 <3,5>", '\000' <repeats 31 times>, "\001\000\000\000\000\000\000\000\000\216\334Ek\214\262Q\360\336\377\377\377\177\000\000 \341\377\377\377\177\000\000\314\373\377\377\377\017\000\000`\336\377\377\377\177\000\000p\341\377\377\377\177\000\000`\336\377\377\377\177\000\000P\341\377\377\377\177\000\000z\033\061VUU\000\000\022LzVUU\000\000\000[X\000\001a\000\000\260\341\377\377\377\177\000\000\340\\ZVUU\000\000 \321`VUU\000\000"...
        the_args = {{gp_offset = 40, fp_offset = 48, overflow_arg_area = 0x7fffffffe3f0, 
            reg_save_area = 0x7fffffffe310}}
#5  0x00005555561b9c09 in timer_sanity_check () at timeout.c:2402
        where = 196613
        x = 3 '\003'
        y = 5 '\005'
        curr = 0x604001568ed0
#6  0x000055555585f746 in sanity_check () at cmd.c:4843
No locals.
#7  0x0000555555763767 in moveloop (resuming=0 '\000') at allmain.c:767
        moveamt = 12
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        timeout_start = 30460
        past_clock = -12056
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#8  0x00005555563bb426 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
<img width="960" alt="issue 23" src="https://user-images.githubusercontent.com/5000096/198130226-95e0bde9-470a-4096-ae0e-5883947a6b69.png">

**Comment by elunna** [comment]
This only popped up after adding Scrolls of Ice

**Comment by elunna** [comment]
Attempted a fix in this commit: 454313de7

Scrolls of ice were not setting melt timers on the ice they created - this patch makes scrolls of ice use the same routine as wands of cold, so hopefully the handling is a bit more refined.
